android/app-newm/src/main/java/io/newm/utils/AndroidNewmAppAnalyticsTracker.kt

```diff
@@ -0,0 +1,128 @@
+package io.newm.utils
+
+import android.os.Bundle
+import com.google.firebase.analytics.FirebaseAnalytics
+import com.google.firebase.analytics.ktx.analytics
+import com.google.firebase.analytics.logEvent
+import com.google.firebase.ktx.Firebase
+import io.newm.shared.AppAnalyticsTracker
+import io.newm.shared.NewmAppLogger
+import org.json.JSONObject
+
+/**
+ * Implementation of [AppAnalyticsTracker] for Android using Firebase Analytics.
+ */
+class AndroidNewmAppAnalyticsTracker(val logger: NewmAppLogger) : AppAnalyticsTracker {
+
+    private val firebaseAnalytics = Firebase.analytics
+
+    companion object {
+        private const val MAX_EVENT_NAME_LENGTH =
+            32 // Firebase allows 40, but using 32 for readability
+        private const val TAG = "AnalyticsTracker" // Tag for logging
+    }
+
+    override fun trackEvent(eventName: String, properties: Map<String, Any?>?) {
+        val safeEventName = validateEventName(eventName)
+        if (safeEventName == null) {
+            logger.info(TAG, "Invalid event name: $eventName. Event not tracked.")
+            return
+        }
+
+        val bundle = Bundle().apply {
```

Comment:
There is an Android Ktx function that already does this. You can simply do here:
```
val bundle = bundleOf(*properties?.toList()?.toTypedArray().orEmpty()) 
```